src/test/java/org/kie/processmigration/test/ContainerKieServerLifecycleManager.java

```diff
@@ -0,0 +1,100 @@
+package org.kie.processmigration.test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.appformer.maven.integration.MavenRepository;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.kie.api.KieServices;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+
+public class ContainerKieServerLifecycleManager implements QuarkusTestResourceLifecycleManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ContainerKieServerLifecycleManager.class);
+    private static final String URL = "http://%s:%s/kie-server/services/rest/server";
+    private static final String CONTAINER_IMAGE = System.getProperty("kieserver.container.name", "jboss/kie-server-showcase");
+    private static final String CONTAINER_TAG = System.getProperty("kieserver.container.tag", "latest");
+
+    public static final String GROUP_ID = "com.myspace.test";
+    public static final String ARTIFACT_ID = "test";
+    public static final String CONTAINER_ID = "test";
+    public static final String KIE_SERVER_ID = "pim-kie-server";
+
+    private final GenericContainer container;
+
+    public ContainerKieServerLifecycleManager() throws IOException {
+        LOGGER.info("Trying to create container for: {}/{}", CONTAINER_IMAGE, CONTAINER_TAG);
+
+        KieServices ks = KieServices.Factory.get();
+        MavenRepository repo = MavenRepository.getMavenRepository();
+        for (String version : List.of("1.0.0", "2.0.0")) {
+            org.kie.api.builder.ReleaseId builderReleaseId = ks.newReleaseId(GROUP_ID, ARTIFACT_ID, version);
+            File kjar = readFile(CONTAINER_ID + "-" + version + ".jar");
+            File pom = readFile(CONTAINER_ID + "-" + version + ".pom");
+            repo.installArtifact(builderReleaseId, kjar, pom);
+        }
+
+        this.container = new GenericContainer(DockerImageName.parse(CONTAINER_IMAGE + ":" + CONTAINER_TAG))
+                .withExposedPorts(8080)
```

Comment:
I would use something similar to the other properties rather than hardcoding the port.
```suggestion
                .withExposedPorts(CONTAINER_PORT)
```

Reply:
@ruromero please see above